c2rust-analyze/src/rewrite/apply.rs

```diff
@@ -361,7 +361,7 @@ impl<S: Sink> Emitter<'_, S> {
                 })
             }
             Rewrite::MethodCall(ref method, ref receiver_rw, ref arg_rws) => {
-                self.emit(receiver_rw, 0)?;
+                self.emit(receiver_rw, 3)?;
```

Comment:
What's the magic 3 value here?

Reply:
Operator precedence, roughly.  Things that bind less tightly than `.` need to be parenthesized when used as a method receiver.  E.g. `(2 + 2).method()`

Reply:
Might be useful to add a comment for it, and/or change it to a `const` with a good name.

Reply:
Changing all the precedence values to named `const`s might be a good idea.  I think that should be a separate PR, though